### PR TITLE
Add a string splitter class

### PR DESCRIPTION
**Pull request by andy31415**
Create a class that can iterate over a character-separated string.
In other PRs, e.g. for argument parsing, I would like to have comma separated value iteration.

Splitting this into its own helper class, with unit tests.

### CODE REVIEW

**Review by vivien-apple — APPROVED** [review]

File: src/lib/support/tests/TestStringSplitter.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+{
+    // empty string handling
+    {
+        StrdupStringSplitter splitter("", ',');
+
+        // next stays at nullptr
+        NL_TEST_ASSERT(inSuite, splitter.Next() == nullptr);
+        NL_TEST_ASSERT(inSuite, splitter.Next() == nullptr);
+        NL_TEST_ASSERT(inSuite, splitter.Next() == nullptr);
+    }
```
[the remark this answers is not in the document]
Comment (andy31415) [reply src/lib/support/tests/TestStringSplitter.cpp:36]: Added tests. Previous code would also not emit a final empty element. Even if that is unusual, fixed it anyway.

**Review by vivien-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]

File: src/lib/support/StringSplitter.h [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+///   "a," split by ',': "a", ""
+///   ",a" split by ',': "", "a"
+///
+///
+/// WARNING: WILL DESTRUCTIVELY MODIFY THE STRING IN PLACE
+///
+class StringSplitter
+{
+public:
+    StringSplitter(char * s, char separator) : mNext(s), mSeparator(separator)
```
Comment (tcarmelveilleux) [inline src/lib/support/StringSplitter.h:42]: This cannot work against a `const char *`. Would recommend CharSpans instead:

- Can work with const strings
- No issues with non-null-terminated source strings

Comment (andy31415) [reply src/lib/support/StringSplitter.h]: That is a great suggestion. Applied.

**Review by robszewczyk — APPROVED** [review]

File: src/lib/support/StringSplitter.h [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+public:
+    StringSplitter(const char * s, char separator) : mNext(s), mSeparator(separator)
+    {
+        if ((mNext != nullptr) && (*mNext == '\0'))
+        {
+            mNext = nullptr; // end of string right away
+        }
+    }
+
+    /// Returns the next character san
```
Comment (bzbarsky-apple) [inline src/lib/support/StringSplitter.h:47]: "span", not "san"?

Comment (andy31415) [reply src/lib/support/StringSplitter.h:47]: oops ... patched into #27156 because small delta.

**Pull request merged**